pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -39,26 +39,18 @@ type Endpoint struct {
 // endpoints `location` from the `ref` and creates a new named reference from it.
 // The function errors if the newly created reference is not parsable.
 func (e *Endpoint) RewriteReference(ref reference.Named, prefix string) (reference.Named, error) {
-	if ref == nil {
```

Comment:
From https://github.com/containers/image/pull/617#discussion_r278058332 by @saschagrunert :
> I'm personally not a fan of removing this check since it is a public API and crashing would be no option for me. But this is just my personal non-blocking opinion.

Reply:
Rationale from the commit message:

> This "should never happen" (famous last words), and even if it did, a panic with a backtrace is more useful than an error returned up the stack and not indicating the exact incorrect caller.

We don’t have many such checks in this repo; a crash (which can, ultimately, still be caught in Go using `recover`) works a bit better for fixing the bug, and this is not C where a risk of silent memory corruption can make users blame the library for being unreliable instead of assigning the blame correctly.

Anyway, I wouldn’t mind dropping this change if others prefer to keep the check.

Reply:
I concur with @mtrmac that such checks are very uncommon in the stack.